src/address_operands.rs

```diff
@@ -74,6 +74,11 @@ pub fn address_operands_read(vm: &mut VMState, opcode: &Opcode) -> (TaggedValue,
                 }
             }
         }
+    };
+    if opcode.swap_flag {
```

Comment:
Could there be a combination of inputs that when swapped becomes non-valid? Eg: If I do `add.swap 1, r0, r1`

Reply:
Compiler forbids this kind of instructions:
```asm
	div.s r1, 3, r3, r4
```
throws:
```bash
error: assembly parsing: can not parse text section element: cannot parse lines: {6: ("div.s r1, 3, r3, r4", InvalidOperandImmInRegLocation("3"))}
```